src/main/java/com/google/cloud/pubsublite/flink/PubsubLiteSourceSettings.java

```diff
@@ -94,6 +95,38 @@
 
   abstract @Nullable SerializableSupplier<CursorClient> cursorClientSupplier();
 
+  abstract @Nullable SerializableSupplier<TopicStatsClient> topicStatsClientSupplier();
+
+  AdminClient getAdminClient() {
+    if (adminClientSupplier() != null) {
+      return adminClientSupplier().get();
+    }
+    return AdminClient.create(
```

Comment:
I think you could provide this as the default value in constructor, then you don't need nullable and you could avoid the null check. Same applies to below methods.

Reply:
Unfortunately this doesn't work :(

This method depends on the subscription path, which isn't available when we set the default values.

Reply:
I see, SG.